server/src/fhir/cacheStructureDefinitions.ts

```diff
@@ -0,0 +1,72 @@
+const nodeCache = require('node-cache')
+const _ = require('lodash')
+
+const cache = new nodeCache()
+
+export const setStructureDefinition = (fhirStructureDefinition: any) => {
+  const struct = buildCustomStructure(fhirStructureDefinition.resource)
+
+  // Use id as key. If it isn't present, use url
+  const key = struct.id || struct.url
+
+  if (key === undefined) {
+    throw new Error('Structure definition has no id nor url field.')
+  }
+
+  // Put obj in key value store
+  cache.set(key, struct)
+}
+
+export const getStructureDefinition = (key: string) => cache.get(key)
+
+const buildCustomStructure = (structResource: any): any => {
```

Comment:
I would rename `buildCustomStructure` to `structuralize` ("that puts into structure") and `structResource` to `definition`. Does that make sense ?